Raw tx pkt sample (#12872)
nrfconnect/sdk-nrf: samples/wifi/raw_tx_packet/README.rst

.. _wifi_raw_tx_packet_sample:

> Wi-Fi: Raw TX Packet

Reviewer: Perhaps "packet" throughout the doc?

####################

.. contents::
   :local:
   :depth: 2

The Raw TX Packet sample demonstrates how to transmit raw TX packets.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview

> ********

Reviewer 2: ```suggestion
********

```

The sample demonstrates how to transmit raw TX packets in both Station-connected and Station-idle modes of operation.
The sample provides the option to transmit raw TX packets either continuously or in a fixed number of packets.

Configuration
*************

|config|

Configuration options
=====================

> The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`) :

Reviewer 2: ```suggestion
The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):
```

.. options-from-kconfig::

> Configuration options for Connected mode

Reviewer: ```suggestion
Configuration options for Connected mode
----------------------------------------
```

> **Configuration options for connected mode:**

Reviewer 2: @D-Triveni I think this suggestion got missed. Please implement when you get the chance.

> To configure the sample in connected mode, you must configure the following Wi-Fi® credentials in the :file:`prj.conf` file:

Reviewer: ```suggestion
To configure the sample in Connected mode, you must configure the following Wi-Fi® credentials in the :file:`prj.conf` file:
```

> To configure the sample in connected mode, you must configure the following Wi-Fi credentials in the :file:`prj.conf` file:

Reviewer: ```suggestion
To configure the sample in connected mode, you must configure the following Wi-Fi® credentials in the :file:`prj.conf` file:
```

* ``Service Set Identifier (SSID)``: Defines the name of your Wi-Fi network.
* ``Key management``: Specifies the key management method, such as WPA2, WPA3, or other encryption protocols.
* ``Password``: Sets the Pre-shared Key (PSK) of your Wi-Fi network.

.. note::

>    You can also use ``menuconfig`` to enable ``Key management`` option.

Reviewer 2: ```suggestion
   You can also use ``menuconfig`` to enable the ``Key management`` option.
```

Reviewer: This got missed.

See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how to run ``menuconfig``.

> Configuration options for Idle mode

Reviewer 2: ```suggestion
Configuration options for Idle mode
----------------------------------------
```

An similar suggestion for line 36

> Configuration options for Idle mode

Reviewer: ```suggestion
Configuration options for Idle mode
-----------------------------------
```

To configure the sample in Idle mode, you must configure the following parameter in the :file:`prj.conf` file:

* ``Channel``: Specifies the Wi-Fi channel to be used for communication on the wireless network.

> Configuration options for Raw TX Packet header

Reviewer: ```suggestion
Configuration options for Raw TX Packet header
----------------------------------------------
```

> * ``Rate value``: Specifies the data transmission rate.

Reviewer 3: ```suggestion
* ``MCS index or data Rate value``: Specifies the data transmission PHY rate.
```

* ``Rate flags``: Specifies the data transmission mode.
* ``Queue number``: Specifies the transmission queue to which raw TX packets are assigned for sending.

Additionally, you must configure the following option in the :file:`prj.conf` file: to define the time delay between raw TX packets:

> ``Inter-frame delay``: Sets the time duration between raw TX packets.

Reviewer 2: ```suggestion
* ``Inter-frame delay``: Sets the time duration between raw TX packets.
```

IP addressing

> *************

Reviewer 2: ```suggestion
*************

```

The sample uses DHCP to obtain an IP address for the Wi-Fi interface.
It starts with a default static IP address to handle networks without DHCP servers, or if the DHCP server is not available.
Successful DHCP handshake will override the default static IP configuration.

You can change the following default static configuration in the :file:`prj.conf` file:

.. code-block:: console

  CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.1.98"
  CONFIG_NET_CONFIG_MY_IPV4_NETMASK="255.255.255.0"
  CONFIG_NET_CONFIG_MY_IPV4_GW="192.168.1.1"

Building and running
********************

.. |sample path| replace:: :file:`samples/wifi/raw_tx_packet`

.. include:: /includes/build_and_run_ns.txt

The sample can be built for the following configurations:

* Continuous transmission with the Station-connected mode.
* Fixed number of packet transmission with the Station-connected mode.
* Continuous transmission with the Station-idle mode.
* Fixed number of packet transmission with the Station-idle mode.

To build for the nRF7002 DK, use the ``nrf7002dk_nrf5340_cpuapp`` build target.
The following are examples of the CLI commands:

> * Continuous transmission with the station-connected mode:

Reviewer 2: ```suggestion
* Continuous transmission with the Station-connected mode:
```

Reviewer: This got missed.

  .. code-block:: console

     west build -b nrf7002dk_nrf5340_cpuapp -- -DCONFIG_RAW_TX_PACKET_SAMPLE_CONNECTION_MODE=y -DCONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE=0

> * Fixed number of packet transmission with the station-connected mode:

Reviewer 2: ```suggestion
* Fixed number of packet transmission with the Station-connected mode:
```

Reviewer: This got missed and at lines 108, 114 as well.

  .. code-block:: console

     west build -b nrf7002dk_nrf5340_cpuapp -- -DCONFIG_RAW_TX_PACKET_SAMPLE_CONNECTION_MODE=y -DCONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE=1 -DCONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS=<number of packets to be sent>

> * Continuous transmission with the station-idle mode:

Reviewer 2: ```suggestion
* Continuous transmission with the Station-idle mode:
```

  .. code-block:: console

     west build -b nrf7002dk_nrf5340_cpuapp -- -DCONFIG_RAW_TX_PACKET_SAMPLE_IDLE_MODE=y -DCONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE=0

> * Fixed number of packet transmission with the station-idle mode:

Reviewer 2: ```suggestion
* Fixed number of packet transmission with the Station-idle mode:
```

  .. code-block:: console

     west build -b nrf7002dk_nrf5340_cpuapp -- -DCONFIG_RAW_TX_PACKET_SAMPLE_IDLE_MODE=y -DCONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE=1 -DCONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS=<number of packets to be sent>

> Change the build target as given below for the nRF7002 EK.

Author: No, this can't be done for this example. All the above build commands can be executed with 'EK' by simply replacing the 'target_board' name.

.. _wifi_raw_tx_packet_sample:

Wi-Fi: Raw TX packet
####################

.. contents::
   :local:
   :depth: 2

The Raw TX packet sample demonstrates how to transmit raw TX packets.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

> The sample demonstrates how to transmit raw TX packets in both Station-connected and Station-idle modes of operation.

Reviewer 4: Maybe good to mention that these different options will be Kconfig-based configurations, not runtime.

The sample provides the option to transmit raw TX packets either continuously or in a fixed number of packets.

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

.. options-from-kconfig::

Configuration options for Connected mode
----------------------------------------

To configure the sample in Connected mode, you must configure the following Wi-Fi® credentials in the :file:`prj.conf` file:

* ``Service Set Identifier (SSID)``: Defines the name of your Wi-Fi network.

> * ``Key management``: Specifies the key management method, such as WPA2, WPA3, or other encryption protocols.

Reviewer: ```suggestion
* ``Key management``: Specifies the key management method, such as WPA2™, WPA3™, or other encryption protocols.
```

.. _wifi_raw_tx_packet_sample:

Wi-Fi: Raw TX packet
####################

.. contents::
   :local:
   :depth: 2

The Raw TX packet sample demonstrates how to transmit raw TX packets.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

The sample demonstrates how to transmit raw TX packets in both connected Station and non-connected Station modes of operation.

> The sample provides the option to transmit raw TX packets either continuously or in a fixed number of packets.

Reviewer 5: can this be made a bulleted point such as 

The sample provides the option to transmit raw TX packets as described below:
1) Continuous packet transmission. 
2) Fixed number of packet transmit.

> The options to select connected Station/non-connected Station mode and continuous/fixed packets are build-time configurations.

Reviewer 5: ```suggestion
The options to select connected Station/non-connected Station mode and continuous/fixed packet transmissions are build-time configurations.
```

Author: Rephrased the sentence.

Configuration
*************

|config|

Configuration options
=====================

> The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

Reviewer 5: ```suggestion
The following sample specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):
```

Author: hyphen should be there.

* ``Rate flags``: Specifies the data transmission mode.
* ``Queue number``: Specifies the transmission queue to which raw TX packets are assigned for sending.

Additionally, you must configure the following option in the :file:`prj.conf` file: to define the time delay between raw TX packets:

* ``Inter-frame delay``: Sets the time duration between raw TX packets.

IP addressing
*************

The sample uses DHCP to obtain an IP address for the Wi-Fi interface.
It starts with a default static IP address to handle networks without DHCP servers, or if the DHCP server is not available.
Successful DHCP handshake will override the default static IP configuration.

You can change the following default static configuration in the :file:`prj.conf` file:

.. code-block:: console

  CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.1.98"
  CONFIG_NET_CONFIG_MY_IPV4_NETMASK="255.255.255.0"
  CONFIG_NET_CONFIG_MY_IPV4_GW="192.168.1.1"

Building and running
********************

.. |sample path| replace:: :file:`samples/wifi/raw_tx_packet`

.. include:: /includes/build_and_run_ns.txt

The sample can be built for the following configurations:

> * Continuous transmission with the connected Station mode.

Reviewer 5: Should this be 

Continuous raw 802.11 packet transmission in connected Station mode.

Similarly for the below sentences. change "with the" to in.
Also, should it be referred that the mode is Station with TX Injection mode

.. _wifi_raw_tx_packet_sample:

Wi-Fi: Raw TX packet
####################

.. contents::
   :local:
   :depth: 2

> The Raw TX packet sample demonstrates how to transmit raw TX packets.

Reviewer 4: ```suggestion
The Raw TX packet sample demonstrates how to transmit raw IEEE 802.11 packets.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

The sample demonstrates how to transmit raw TX packets in both connected Station and non-connected Station modes of operation.

> The sample provides the option to select the either of the modes of transmission:

Reviewer 4: ```suggestion
The sample provides the option to select the traffic pattern between the following modes:
```

* Continuous packet transmission.

> * Fixed number of packet transmission.

Reviewer 4: ```suggestion
* Fixed number of transmitted packets.
```

The configurations for connected Station or non-connected Station modes, and for continuous or fixed packet transmission, are set at build time.

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

.. options-from-kconfig::

> Configuration options for connected Station mode

Reviewer 4: Could we add some indentation here, for ease of reading? That is to ensure that the reader understand what configuration is needed in each case:

- Operating mode
  - Configuration options for connected Station mode
  - Configuration options for non-connected Station mode
- Configuration options for Raw TX packet header
- Misc (or, Additional configurations)

.. _wifi_raw_tx_packet_sample:

Wi-Fi: Raw TX packet
####################

.. contents::
   :local:
   :depth: 2

The Raw TX packet sample demonstrates how to transmit raw IEEE 802.11 packets.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********
The sample generates and broadcasts 802.11 beacon frames as raw TX packets.
As a consequence, the nRF70 Series device can be identified as a Wi-Fi® beaconing device.

The sample demonstrates how to transmit raw TX packets in both connected Station and non-connected Station modes of operation.
The sample provides the option to select the traffic pattern between the following modes:

> * Continuous packet transmission.

Reviewer 3: Refer Kconfig options `:ref:kconfig:option ...`

* Fixed number of transmitted packets.

The configurations for connected Station or non-connected Station modes, and for continuous or fixed packet transmission, are set at build time.

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

.. options-from-kconfig::

By using the following configuration options, you can configure the sample for different operational modes, including specific raw TX packet header settings:

Configuration options for operating modes
-----------------------------------------

  * For connected Station mode

    To configure the sample in connected Station mode, you must configure the following Wi-Fi credentials in the :file:`prj.conf` file:

>     * ``Service Set Identifier (SSID)``: Defines the name of your Wi-Fi network.

Reviewer 3: please use `:ref:kconfig:option ...`

Reviewer: We could also use :ref: to link to the Kconfig options which are mentioned in the previous section "Configuration options".
For example: see this: https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-12872/nrf/samples/wifi/sr_coex/README.html#configuration-files

    * ``Password``: Sets the Pre-shared Key (PSK) of your Wi-Fi network.

    .. note::
       You can also use ``menuconfig`` to enable the ``Key management`` option.

    See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how to run ``menuconfig``.

  * For non-connected Station mode

    To configure the sample in non-connected Station mode, you must configure the ``Channel`` parameter in the :file:`prj.conf` file.

    This specifies the Wi-Fi channel to be used for communication on the wireless network.

Configuration options for raw TX packet header
----------------------------------------------

  The following configuration options are available for the raw TX packet header:

  * ``MCS index or Data rate value``: Specifies the data transmission PHY rate.
  * ``Rate flags``: Specifies the data transmission mode.
  * ``Queue number``: Specifies the transmission queue to which raw TX packets are assigned for sending.

Additionally, you must configure the ``Inter-frame delay`` parameter in the :file:`prj.conf` file to define the time delay between raw TX packets.

This sets the time duration between raw TX packets.

IP addressing
*************

> The sample uses DHCP to obtain an IP address for the Wi-Fi interface.

Reviewer 3: please add a note that this is only for connected mode.

.. _wifi_raw_tx_packet_sample:

Wi-Fi: Raw TX packet
####################

.. contents::
   :local:
   :depth: 2

The Raw TX packet sample demonstrates how to transmit raw IEEE 802.11 packets.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********
The sample generates and broadcasts 802.11 beacon frames as raw TX packets.

> As a consequence, the nRF70 Series device can be identified as a Wi-Fi beaconing device.

Reviewer: ```suggestion
As a consequence, the nRF70 Series device can be identified as a Wi-Fi® beaconing device.
```

The sample demonstrates how to transmit raw TX packets in both connected Station and non-connected Station modes of operation.
The sample provides the option to select the traffic pattern between the following modes:

* Continuous packet transmission.
* Fixed number of transmitted packets.

The configurations for connected Station or non-connected Station modes, and for continuous or fixed packet transmission, are set at build time.

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

.. options-from-kconfig::

> 

Reviewer: We can add a line here like, "Additionally, the following configuration options are available for the sample:"
Then we can move the content under these subheadings to avoid redundancy.

* Operating mode

  * Configuration options for connected Station mode
  * Configuration options for non-connected Station mode

* Configuration options for Raw TX packet header

Configuration options for connected Station mode
------------------------------------------------

> To configure the sample in connected Station mode, you must configure the following Wi-Fi® credentials in the :file:`prj.conf` file:

Reviewer: ```suggestion
To configure the sample in connected Station mode, you must configure the following Wi-Fi credentials in the :file:`prj.conf` file:
```

The configurations for connected Station or non-connected Station modes, and for continuous or fixed packet transmission, are set at build time.

Configuration
*************

|config|

Configuration options
=====================

The following sample-specific Kconfig options are used in this sample (located in :file:`samples/wifi/raw_tx_packet/Kconfig`):

.. options-from-kconfig::

Configuration options for operating modes
-----------------------------------------

By using the following Kconfig options, you can configure the sample for different operational modes:

* For connected Station mode

  To configure the sample in connected Station mode, you must configure the following Wi-Fi credentials in the :file:`prj.conf` file:

  * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_SSID`: Sets the name of your Wi-Fi network.
  * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_NONE`: Selects open security.
  * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2`: Selects PSK security.
  * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2_256`: Selects PSK-256 security.

>   * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA3`: Selects SAE security(default).

Reviewer: ```suggestion
  * :kconfig:option:`CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA3`: Selects SAE security (default).
```
